src/Rule/Nested.php

```diff
@@ -169,4 +168,36 @@ private function fetchOptions(array $rules): array
 
         return $result;
     }
+
+    private static function calculateErrorKey(Rules $aggregatedRule, string $valuePath, string $key): string
+    {
+        if (!self::canConcatenateErrorKey($aggregatedRule, $key)) {
+            return $valuePath;
+        }
+
+        return $valuePath . Result::ERROR_KEY_SEPARATOR . $key;
+    }
+
+    private static function canConcatenateErrorKey(Rules $aggregatedRule, string $key): bool
+    {
+        if (!self::isPositiveInteger($key)) {
+            return true;
+        }
+
+        $aggregatedRuleArr = $aggregatedRule->asArray();
+        if (ArrayHelper::getValueByPath($aggregatedRuleArr, '0.0') !== 'each') {
+            return false;
+        }
+
+        return !(ArrayHelper::getValueByPath($aggregatedRuleArr, '0.1.0') === 'nested');
+    }
+
+    private static function isPositiveInteger(string $str): bool
```

Comment:
Maybe add this method to Yii Strings: https://github.com/yiisoft/strings/blob/master/src/NumericHelper.php ?

Reply:
Yep, sounds like a good idea. Thought about adding this to a more generic helper within the current repository. But indeed, we can add this to `NumericHelper`. Will do later.

Reply:
https://github.com/yiisoft/strings/pull/78

Reply:
:heavy_check_mark:

Reply:
Also adjusted to not consider 0 as positive integer.

Reply:
Tests are now failing in CI because we need to release https://github.com/yiisoft/strings/pull/78 first. Tested localy with new changes, they are passing.

Reply:
I simplified and refactored implementation and it's no longer needed here. Still can be useful in strings package.

Reply:
With new approach (using `Error` objects) there is no check for concatenation anymore, so this is no longer actual.